### PR TITLE
More improvements to template.

### PR DESCRIPTION
**Pull request by Raynos**
See commits one by one for reviewing and details.

reviewers: @kriskowal @malandrew @Matt-Esch

cc: @sh1mmer

### CODE REVIEW

**Comment by sh1mmer** [comment]
Can you add a test to complain if the README isn't updated?

File: templates/uber/content/package.json [hunk 1]
```diff
@@ -5,43 +5,49 @@
...
+    "test": "npm run check-ls -s && npm run lint -s && npm run cover -s",
+    "test-cover": "istanbul cover --report html --print detail -- test/index.js",
+    "trace": "itape test/index.js --trace",
     "travis": "npm run cover -s && istanbul report lcov && ((cat coverage/lcov.info | coveralls) || exit 0)",
-    "unit-test": "node test/index.js | tap-spec",
     "view-cover": "opn ./coverage/index.html"
   },
-  "engine": {
+  "engines": {
     "node": ">= 0.8.x"
```
Comment (sh1mmer) [inline templates/uber/content/package.json]: Should this be 0.10.x now?

File: templates/uber/index.js [hunk 2]
```diff
@@ -60,6 +60,9 @@ module.exports = {
     },
     gitName: fetchFromGitConfig('user.name'),
     email: fetchFromGitConfig('user.email'),
+    projectNoDash: function readProjectNoDash(values, cb) {
```
Comment (sh1mmer) [inline templates/uber/index.js]: if we are doing no dash do we also want to normalize case?

**Comment by Raynos** [comment]
@sh1mmer I've considered adding a readme linter.

I searched for available packages to lint either readmes or markdown. I could only find [readme-good](https://github.com/btford/readme-good) which doesn't solve our need.

I would like to write a README linter, however it's non-trivial. I think adding it is out of scope for this PR, but if I have time I'll come back to it late next week.

File: templates/uber/content/package.json (hunk 1 above)
Comment (Raynos) [inline templates/uber/content/package.json]: good idea.

**Comment by sh1mmer** [comment]
@Raynos I think there is a difference between something applying a heuristic on the "goodness" of a README vs. something that just checks it's been updated from what was scaffolded.

E.g. I'd like to make the project with a failing test because the README needs to be updated to make it pass (and then you would delete the test one assumes).

**Comment by Raynos** [comment]
@sh1mmer That's an interesting idea.

I'm worried about the user experience for that approach. Having a failing test automatically in your project that you have to delete feels a bit weird. It would also be hard to communicate what fails with a test.

The user experience would be better if we have a linter that checks your README, for a few reasons

 - People tend to write code & tests before the README, they want their tests to pass
 - The linter can output warnings or errors.
 - The linter is seperate from the test suite, people can write code & tests without fixing linting errors.

If we add checks I want to make sure it does not get in the way. I'd rather do it with a good user experience rather then bolt something on.

**Comment by sh1mmer** [comment]
I think the linter could do the warning for sure. I'd be happy to put a "UPDATE THE README TO REMOVE THIS LINE TO MAKE THE LINTER PASS" or something in the README.md to make that trivial.

It would be nice to have a universal solution to this that isn't dependent on Uber internal tools for enforcement and works I equally well for open source. I figure a failing test is just TDD style and would work universally well for anything.

If the linter is part of the test suite I think that's very acceptable for this criteria.

**Comment by sh1mmer** [comment]
Actually one thought that just occurs is that if we can assume git could have a test/lint that checks you updated the README if you update a version. That would work universally for new or existing projects. It's probably out of scope for this diff though, so I'd prefer to start with a simple solution.

**Comment by sh1mmer** [comment]
Other than my one comment LGTM.

**Comment by Raynos** [comment]
Noted, we need a README linter.

We can write a simple one that checks there are no TODOs in the README itself.

I don't want to block this particular PR to wait for that module though, so I'll circle back to it later this week if I have time.

**Comment by sh1mmer** [comment]
I agree we shouldn't block on a README linter.

**Pull request merged**